### PR TITLE
[AIRFLOW-6545] Add commit message validation

### PR DESCRIPTION
**Pull request by potiuk**
Adds validation of AIRFLOW* jira issue id to all commits.

Note that it requires https://github.com/kaxil/boring-cyborg/pull/2 to be merged in @kaxil 's boring-cyborg and it requires to grant new permissions to the boring-cyborg: `Commit status: read&write.`

---
Issue link: [AIRFLOW-6545](https://issues.apache.org/jira/browse/AIRFLOW-6545/)

- [x] Description above provides context of the change
- [x] Commit message/PR title starts with `[AIRFLOW-NNNN]`. AIRFLOW-NNNN = JIRA ID<sup>*</sup>
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

<sup>*</sup> For document-only changes commit message can start with `[AIRFLOW-XXXX]`.

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Review by turbaszek — APPROVED** [review]

File: .github/boring-cyborg.yml [hunk 1]
```diff
@@ -182,3 +182,7 @@ insertIssueLinkInPrDescription:
     docOnlyIssueMatch:
       titleIssueIdRegexp: \[(AIRFLOW-X{4})\]
       descriptionIssueLink: "`Document only change, no JIRA issue`"
+
+
+verifyCommitMessages:
+  commitMessageRegexp: ^.*\[AIRFLOW-[0-9]{4}\].*$|^.*\[AIRFLOW-[X]{4}\].*$
```
Comment (turbaszek) [inline .github/boring-cyborg.yml]:
```suggestion
  commitMessageRegexp: ^.*\[AIRFLOW-[0-9]{4}\].*$|^.*\[AIRFLOW-XXXX\].*$
```

**Comment by potiuk** [comment]
Hey @kaxil  - do not merge it yet as I want to make some small update after talking to @nuclearpinguin and @mik-laj

**Comment by potiuk** [comment]
OK. Ready to merge. I improved the https://github.com/kaxil/boring-cyborg/pull/2 to better handle multiple/single commit cases and updated regexps :)

**Review by kaxil — APPROVED** [review]

**Comment by kaxil** [comment]
Awesome, @potiuk

**Pull request merged**